libraries/aws_rds_instance.rb

```diff
@@ -0,0 +1,75 @@
+require '_aws'
+
+class AwsRdsInstance < Inspec.resource(1)
+  name 'aws_rds_instance'
+  desc 'Verifies settings for an rds instance'
+  example "
+    describe aws_rds_instance(db_instance_identifier: 'test-instance-id') do
+      it { should exist }
+    end
+  "
+
+  include AwsResourceMixin
+  attr_reader :db_id
+
+  def to_s
+    "RDS Instance #{@db_id}"
+  end
+
+  private
+
+  def validate_params(raw_params)
+    validated_params = check_resource_param_names(
+      raw_params: raw_params,
+      allowed_params: [:db_id],
+      allowed_scalar_name: :db_id,
+      allowed_scalar_type: String,
+    )
+    if validated_params.empty? or !validated_params.key?(:db_id)
+      raise ArgumentError, 'You must provide an id for the aws_rds_instance.'
+    end
+
+    if validated_params.key?(:db_id) && validated_params[:db_id] !~ /^[a-z]{1}[0-9a-z\-]{0,62}$/
+      raise ArgumentError, 'aws_rds_instance Database Instance ID must be in the format: start with a letter followed by up to 62 letters/numbers/hyphens.'
+    end
+
+    validated_params
+  end
+
+  def fetch_from_aws
+    backend = AwsRdsInstance::BackendFactory.create
+
+    # Transform into filter format expected by AWS
```

Comment:
This block - from line 42 to line 48 - is unused, and can be deleted.  It's copied from other resources in which the AWS SDK expects a `filter` option, for which we have to construct criteria from our instance variables.

While describe_db_instances does take a filter option, it's not being used here, and we should remove code that is unused.  We can easily add it back in if we expand the interface.